lib/galaxy/tools/__init__.py

```diff
@@ -2541,7 +2541,7 @@ def iterate(group_inputs, inputs, state_inputs, other_values=None):
             other_values = ExpressionContext( state_inputs, other_values )
             for input_index, input in enumerate( inputs.itervalues() ):
                 # create model dictionary
-                tool_dict = input.to_dict(trans)
+                tool_dict = input.to_dict(trans, other_values=other_values)
```

Comment:
Anytime. So #1037 is legacy code. other_values are already passed a few lines below the above change: https://github.com/galaxyproject/galaxy/blob/dev/lib/galaxy/tools/__init__.py#L2575, so the dynamic fields should work properly without changes. Multiple dependencies with dynamic select fields as described in https://trello.com/c/NOmjJxCi/428-305-potential-bug-in-dynamic-options-filters work. Is there a tool xml which illustrates the issue?